cmake/CMakeLists.txt

```diff
@@ -1324,27 +1325,65 @@ endif()
 
 # TVM
 if (onnxruntime_USE_TVM)
+  if (NOT TARGET tvm)
+    message(STATUS "Include TVM(*).")
+    include(tvm)
+  endif()
   if (onnxruntime_USE_CUDA)
-    set(USE_CUDA ON)
+    if (onnxruntime_USE_STVM)
+      set(USE_CUDA ${onnxruntime_CUDA_HOME} CACHE BOOL "Only defined for TVM")
+      set(USE_MKLDNN ON CACHE BOOL "Only defined for TVM")
+      set(USE_CUDNN ON CACHE BOOL "Only defined for TVM")
+    endif()
+    if (onnxruntime_USE_NUPHAR)
+      set(USE_CUDA ON CACHE BOOL "Only defined for TVM")
+    endif()
   endif()
   if (onnxruntime_USE_LLVM)
-    set(USE_LLVM ON)
+    set(USE_LLVM ON CACHE BOOL "Only defined for TVM")
     add_definitions(-DUSE_TVM_WITH_LLVM)
   endif()
 
-  add_subdirectory(${PROJECT_SOURCE_DIR}/external/tvm EXCLUDE_FROM_ALL)
-  set_target_properties(tvm PROPERTIES FOLDER "External/tvm")
-  set_target_properties(tvm_topi PROPERTIES FOLDER "External/tvm")
-  set_target_properties(tvm_runtime PROPERTIES FOLDER "External/tvm")
+  if (onnxruntime_USE_STVM)
+    set(USE_OPENMP gnu CACHE STRING "Only defined for TVM")
+    set(USE_MICRO OFF CACHE BOOL "Only defined for TVM")
+  endif()
+  message(STATUS "TVM BEFORE USE_LLVM=${USE_LLVM} USE_OPENMP=${USE_OPENMP} USE_MICRO=${USE_MICRO} CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} USE_CUDA=${USE_CUDA} USE_GTEST=${USE_GTEST} CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+  message(STATUS "tvm_SOURCE_DIR=${tvm_SOURCE_DIR}")
+  message(STATUS "tvm_BINARY_DIR=${tvm_BINARY_DIR}")
+  add_subdirectory(${tvm_SOURCE_DIR} ${tvm_BINARY_DIR} EXCLUDE_FROM_ALL)
+  message(STATUS "TVM AFTER USE_LLVM=${USE_LLVM} USE_OPENMP=${USE_OPENMP} USE_MICRO=${USE_MICRO} CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} USE_CUDA=${USE_CUDA} USE_GTEST=${USE_GTEST} CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+
+  if (onnxruntime_USE_NUPHAR)
+    set_target_properties(tvm PROPERTIES FOLDER ${tvm_SOURCE_DIR})
+    set_target_properties(tvm_topi PROPERTIES FOLDER ${tvm_SOURCE_DIR})
+    set_target_properties(tvm_runtime PROPERTIES FOLDER ${tvm_SOURCE_DIR})
+
+    set(TVM_INCLUDES ${tvm_SOURCE_DIR}/include
+      ${tvm_SOURCE_DIR}/3rdparty/dmlc-core/include
```

Comment:
So, please also add dmlc-core and dlpack to our cgmanifest.json. In the json file you may add a comment saying they are for TVM.